erpnext/selling/doctype/customer/customer.py

```diff
@@ -104,9 +104,8 @@ def update_lead_status(self):
 		if self.lead_name:
 			frappe.db.set_value('Lead', self.lead_name, 'status', 'Converted', update_modified=False)
 
-			for doctype in ('Opportunity', 'Quotation'):
-				for d in frappe.get_all(doctype, {'lead': self.lead_name}):
-					frappe.db.set_value(doctype, d.name, 'customer', self.name, update_modified=False)
+		for d in frappe.get_all('Opportunity', {'lead': self.lead_name}):
```

Comment:
This should be inside the if check. If there is no lead name, the `get_all` will return ALL opportunities.